### DISCUSSION TITLE
Refreshing page loads new plot messages faster than adding them

### DISCUSSION THREAD

**Discussion by jameskuszmaul-brt** [foxglove/community#856] [Q&A]
Taking the same MCAP file as https://github.com/orgs/foxglove/discussions/855 (https://drive.google.com/file/d/1a5hTuSFBAJl5pgK3-fR0l2MfQyUu8Nf6/view?usp=sharing), and the below layout as reference:

[Slow plot line adding.json](https://github.com/foxglove/community/files/13044678/Slow.plot.line.adding.json)

On my laptop if I go to the plot panel and add a line for the `"/aos aos.JoystickState".enabled` signal, it takes ~20 seconds to fully render. If I then reload the entire page (e.g. `Ctrl-R` in your normal browser), with the enabled signal now in the saved layout, it only takes ~10 seconds to render all the plot data from when I open the file again. This behavior seems.... odd.

**Comment by cfoust** [comment]
Will investigate! Thank you for reporting.

**Reply by defunctzombie** [reply]
@cfoust did we investigate this? Is there an internal ticket for this or was it a 1-off event that we could not reproduce?

**Reply by cfoust** [reply]
we have an internal ticket; and indeed, I had started to investigate it at the end of the day on Friday!

**Comment by cfoust** [comment]
I'm not able to reproduce this. I tried a range of contrived and user-provided data (including yours) and the load time was always the same, but it may vary by device. I will keep thinking about this, but in the meantime I'm going to close this.
